src/chapters/unit-testing/junit.rst

```diff
@@ -0,0 +1,172 @@
+
+.. sourcecode:: java
+   :linenos:
+
+   import org.junit.Test;
+   import static org.junit.Assert.assertEquals;
+
+If these have not already been added to your file, add them now. As we mention above,
+``@Test`` annotates the method to signal it as a test case. We need to add the appropriate
+classpath in order to take advantage of this annotation. Our empty test is aptly named
+``emptyTest()``, a description of its role. This test does not follow the AAA rule from 
+our :ref:`testing-best-practices`, as it jumps straight to asserting. Nor is it relevant,
+for that matter. Again, the IDE comes in handy, inserting the names of each of our parameters,
+"expected:", "actual:", and "delta:". This empty test is simply asserting an expected 
+value of ``10`` to equal an actual value of ``10``, with an accepted ``.001`` variance.
+Of course, ``10`` equals ``10``. But let's run it so we know our test runner works. Click
+the green arrow to the left of ``public class CarTest`` to run the test. Once run, you'll
+see a new output panel with a green check mark indicating the test passed and a message 
+stating the test passed. We know now how the test runner behaves when a test passes and 
+begin the real work of unit testing the ``Car`` class.
+
+Under your second TODO, write a test to verify that the constructor sets the 
+``gasTankLevel`` property.
+
+.. sourcecode:: java
+   :linenos:
+
+   //TODO: constructor sets gasTankLevel properly
+   @Test
+   public void testInitialGasTank() {
+      Car test_car = new Car("Toyota", "Prius", 10, 50);
+      assertEquals(10, test_car.getGasTankLevel(), .001);
+   }
+
+Here, we give the test a descriptive name, ``testInitialGasTank()``, initialized a new 
+``Car`` object, and test that the constructor correctly handles the ``gasTankLevel`` property.
+By now, you've probably already imported the ``Car`` class.
+
+.. sourcecode:: java
+
+   import org.launchcode.java.demos.testing.main.Car;
+
+Run ``CarTest`` to see that both tests pass. 
+
+.. tip::
+
+   If you want to run only one test, click the green arrow next to the test method's name.
+
+``@Before``
+^^^^^^^^^^^
+
+``@Before`` is another annotation we can use to help in test cases. The ``@Before``
+annotation can be used to set up some data or a condition that you want to have for 
+every test in a given class. In the case of ``CarTest``, it would be nice to not need to
+create a new ``Car`` instance for each test we write. 
+
+In your ``testInitialGasTank()`` method, remove the line initiating ``test_car``. 
+Above your first test, add the following ``@Before`` method:
+
+.. sourcecode:: java
+   :linenos:
+
+   Car test_car;
+
+   @Before
+   public void createCarObject() {
+      test_car = new Car("Toyota", "Prius", 10, 50);
+   }
+
+Did IntelliJ prompt you to import ``@Before``? Did it import the annotation as you were 
+writing the method? We'll assume you know to check what you need to add to your classpath
```

Comment:
I cannot clearly state why, but the sentence "We'll assume..." splashed some cold water on my mood.  Maybe it's a "you should know this by now" vibe?